lib/olly_gc_stats/olly_gc_stats.ml

```diff
@@ -2,23 +2,30 @@ module H = Hdr_histogram
 module Ts = Runtime_events.Timestamp
 
 type ts = { mutable start_time : float; mutable end_time : float }
-
-let total_gc_time = ref 0
 let wall_time = { start_time = 0.; end_time = 0. }
-let total_cpu_time = ref 0.
+
+let domain_cpu_times = Array.make 128 (0.)
 let domain_gc_times = Array.make 128 0
 
-let lifecycle _domain_id _ts lifecycle_event _data =
+let lifecycle domain_id _ts lifecycle_event _data =
   match lifecycle_event with
-  | Runtime_events.EV_RING_START -> wall_time.start_time <- Unix.gettimeofday ()
+  | Runtime_events.EV_RING_START ->
+      let ts = Unix.gettimeofday () in
+      wall_time.start_time <- ts;
+      domain_cpu_times.(domain_id) <- ts
   | Runtime_events.EV_RING_STOP ->
-      wall_time.end_time <- Unix.gettimeofday ();
-      let times = Unix.times () in
-      total_cpu_time := times.tms_utime +. times.tms_cutime
+      let ts = Unix.gettimeofday () in
+      wall_time.end_time <- ts;
+      domain_cpu_times.(domain_id) <- ts -. domain_cpu_times.(domain_id)
+  | Runtime_events.EV_DOMAIN_SPAWN ->
+      domain_cpu_times.(domain_id) <- Unix.gettimeofday ()
+  | Runtime_events.EV_DOMAIN_TERMINATE ->
+      domain_cpu_times.(domain_id) <-
```

Comment:
I originally named this, but on second thought, should we perhaps call it `domain_elapsed_time` since we're recording the wall time?

Reply:
Fixed.